packages/analytics/src/index.native.ts

```diff
@@ -0,0 +1,29 @@
+import { createClient } from '@segment/analytics-react-native';
+import { AnalyticsClientConfig, GenericAnalyticsClient } from 'src/utils/types.shared';
+
+import { AnalyticsClient } from './client/client';
+
+let analyticsClient: AnalyticsClient<GenericAnalyticsClient>;
+
+/**
+ * Configures the analytics client for the native environment. Must be called before any analytics functions are used.
+ * @param {AnalyticsClientConfig} config - The configuration for the analytics client.
+ * @returns {AnalyticsClient<SegmentClient>} The configured analytics client.
+ */
+const configureAnalyticsClient = ({
+  writeKey,
+  defaultProperties,
+  client,
+}: AnalyticsClientConfig) => {
+  const nativeClient =
+    client ??
+    createClient({
+      writeKey,
+    });
+
+  analyticsClient = new AnalyticsClient(nativeClient, { defaultProperties });
+
+  return analyticsClient;
+};
+
+export { analyticsClient, configureAnalyticsClient };
```

Comment:
```suggestion
export function configureAnalyticsClient ({
  writeKey,
  defaultProperties,
  client,
}: AnalyticsClientConfig) {
  const nativeClient =
    client ??
    createClient({
      writeKey,
    });

  analyticsClient = new AnalyticsClient(nativeClient, { defaultProperties });

  return analyticsClient;
}
```

Convention used elsewhere is to use [function declarations](https://www.notion.so/trustmachines/Function-declarations-22f7055df3124d868a242adb04f0f60d?pvs=4)